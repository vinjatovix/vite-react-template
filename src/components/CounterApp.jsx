import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [countValue, setCountValue] = useState(value);
  const handleAdd = () => {
    setCountValue(countValue + 1);
  };
  const handleSubtract = () => {
    setCountValue(countValue - 1);
  };
  const handleReset = () => {
    setCountValue(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2 data-testid="count-value">{countValue}</h2>
      <button aria-label="subtract" onClick={handleSubtract}>
        -1
      </button>
      <button aria-label="reset" onClick={handleReset}>
        Reset
      </button>
      <button aria-label="add" onClick={handleAdd}>
        +1
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};
