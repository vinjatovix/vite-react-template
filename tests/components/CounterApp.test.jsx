import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../../src/components/CounterApp";

describe("CounterApp", () => {
  it("should render correctly", () => {
    const title = "CounterApp";
    const defaultValue = "0";

    const { container } = render(<CounterApp />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
    expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
      defaultValue
    );
    expect(container).toMatchSnapshot();
  });

  it("should render the initial value of 100", () => {
    const value = 100;

    const { getByTestId } = render(<CounterApp value={value} />);

    expect(getByTestId("count-value").innerHTML).toContain(value.toString());
  });

  it("should increment the counter by 1", () => {
    const { getByTestId } = render(<CounterApp />);

    fireEvent.click(screen.getByText("+1"));

    expect(getByTestId("count-value").innerHTML).toContain("1");
  });

  it("should decrement the counter by 1", () => {
    const { getByTestId } = render(<CounterApp />);
    const subtractButton = screen.getByRole("button", { name: "subtract" });

    fireEvent.click(subtractButton);

    expect(getByTestId("count-value").innerHTML).toContain("-1");
  });

  it("should reset the counter to 0", () => {
    const { getByTestId } = render(<CounterApp />);
    const addButton = screen.getByRole("button", { name: "add" });
    const resetButton = screen.getByRole("button", { name: "reset" });
    fireEvent.click(addButton);
    fireEvent.click(addButton);

    fireEvent.click(resetButton);

    expect(getByTestId("count-value").innerHTML).toContain("0");
  });
});
