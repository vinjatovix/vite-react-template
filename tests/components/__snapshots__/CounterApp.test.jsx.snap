// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should render correctly 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="count-value"
  >
    0
  </h2>
  <button
    aria-label="subtract"
  >
    -1
  </button>
  <button
    aria-label="reset"
  >
    Reset
  </button>
  <button
    aria-label="add"
  >
    +1
  </button>
</div>
`;
